scripts/sentiment_analysis/sentiment_analysis_cnn.py

```diff
@@ -0,0 +1,318 @@
+elif args.data_name == 'SST-1':
+    train_dataset, test_dataset = [nlp.data.SST_1(root='data/sst-1', segment=segment)
+                                   for segment in ('train', 'test')]
+    output_size = 5
+elif args.data_name == 'SST-2':
+    train_dataset, test_dataset = [nlp.data.SST_2(root='data/sst-2', segment=segment)
+                                   for segment in ('train', 'test')]
+    output_size = 2
+elif args.data_name == 'Subj':
+    train_dataset, test_dataset = [nlp.data.SUBJ(root='data/Subj', segment=segment)
+                                   for segment in ('train', 'test')]
+    output_size = 2
+elif args.data_name == 'TREC':
+    train_dataset, test_dataset = [nlp.data.TREC(root='data/trec', segment=segment)
+                                   for segment in ('train', 'test')]
+    output_size = 6
+else:
+    print('Unable to find data set')
+    exit()
+
+train_dataset, valid_dataset = nlp.data.train_valid_split(train_dataset, args.valid_ratio)
+
+all_token = []
+max_len = 0
+for line in train_dataset:
+    line = line[0].split(' ')
+    max_len = max_len if max_len > len(line) else len(line)
+    all_token.extend(line)
+print(max_len)
+length_clip = nlp.data.ClipSequence(max_len)
+vocab = nlp.Vocab(nlp.data.count_tokens(all_token))
+vocab.set_embedding(nlp.embedding.create('Word2Vec', source='GoogleNews-vectors-negative300'))
+print(len(vocab))
+net = SentimentNet(dropout=args.dropout, vocab_size=len(vocab))
+
+print('Tokenize using spaCy...')
+
+# Dataset preprocessing
+def preprocess(x):
+    data, label = x
+    data = vocab[length_clip(data.split(' '))]
+    return data, label
+
+def get_length(x):
+    return float(len(x[0]))
+
+def preprocess_dataset(dataset):
+    start = time.time()
+    pool = mp.Pool(8)
+    dataset = gluon.data.SimpleDataset(pool.map(preprocess, dataset))
+    lengths = gluon.data.SimpleDataset(pool.map(get_length, dataset))
+    end = time.time()
+    print('Done! Tokenizing Time={:.2f}s, #Sentences={}'.format(end - start, len(dataset)))
+    return dataset, lengths
+
+# Preprocess the dataset
+train_dataset, train_data_lengths = preprocess_dataset(train_dataset)
+valid_dataset, valid_data_lengths = preprocess_dataset(valid_dataset)
+test_dataset, test_data_lengths = preprocess_dataset(test_dataset)
+
+# Construct the DataLoader. Pad data and stack label
+batchify_fn = nlp.data.batchify.Tuple(nlp.data.batchify.Pad(axis=0, ret_length=True),
+                                      nlp.data.batchify.Stack(dtype='float32'))
+
+train_dataloader = DataLoader(dataset=train_dataset,
+                              batch_size=args.batch_size,
+                              shuffle=True,
+                              batchify_fn=batchify_fn)
+
+valid_dataloader = DataLoader(dataset=valid_dataset,
+                              batch_size=args.batch_size,
+                              shuffle=False,
+                              #sampler=nlp.data.SortedSampler(valid_data_lengths),
```

Comment:
We don't need valid dataset. Use all training dataset to reproduce experimental results.